internal/clients/kafka/acl/acl.go

```diff
@@ -0,0 +1,233 @@
+package acl
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"strings"
+
+	"github.com/crossplane-contrib/provider-kafka/apis/acl/v1alpha1"
+
+	"github.com/pkg/errors"
+	"github.com/twmb/franz-go/pkg/kadm"
+	"github.com/twmb/franz-go/pkg/kmsg"
+)
+
+// AccessControlList is a holistic representation of a Kafka ACL with configurable
+// fields
+type AccessControlList struct {
+	ResourceName                      string
+	ResourceType              string `json:"ResourceType"`
+	ResourcePrinciple                 string `json:"ResourcePrinciple"`
+	ResourceHost                      string `json:"ResourceHost"`
+	ResourceOperation                 string `json:"ResourceOperation"`
+	ResourcePermissionType            string `json:"ResourcePermissionType"`
+	ResourcePatternTypeFilter string `json:"ResourcePatternTypeFilter"`
+}
+
+// List lists all the ACLs in Kafka
+func List(ctx context.Context, cl *kadm.Client, accessControlList *AccessControlList) (*AccessControlList, error) {
+
+	o, err := kmsg.ParseACLOperation(strings.ToLower(accessControlList.ResourceOperation))
+	if err != nil {
+		return nil, errors.Wrap(err, "did not return ACL Operation")
+	}
+
+	ao := []kadm.ACLOperation{o}
+
+	rpt, err := kmsg.ParseACLResourcePatternType(strings.ToLower(accessControlList.ResourcePatternTypeFilter))
+	if err != nil {
+		return nil, errors.Wrap(err, "did not return parsing of ACL pattern")
+	}
+
+	b := kadm.ACLBuilder{}
+	ab := b.Allow(accessControlList.ResourcePrinciple).AllowHosts(accessControlList.ResourceHost).Operations(ao[0]).ResourcePatternType(rpt)
```

Comment:
Is there a case where  `ao[0]` doesn't exist?

Reply:
Not that we've found, no, which is why there's no check here.

Reply:
We don't have a situation where ao[0] would not exist fortunately